umh-core/pkg/config/manager.go

```diff
@@ -142,6 +142,10 @@ type FileConfigManager struct {
 
 	// ---------- in-memory cache (read-only after RLock) ----------
 	cacheMu sync.RWMutex // guards the two fields below
+
+	// ---------- background refresh state ----------
+	refreshMu         sync.Mutex // guards refreshInProgress
+	refreshInProgress bool       // true if a background refresh goroutine is running
```

Comment:
You don't need this variable if you already have a mutex. You can simply call the mutex whenever you start the goroutine